onnxruntime/core/providers/cuda/fpgeneric.cu

```diff
@@ -28,8 +28,10 @@ __global__ void transposeNoOverlap(half* odata, const half* idata, const int m,
   int x = blockIdx.x * TRANS_TILE_DIM + threadIdx.x;
   int y = blockIdx.y * TRANS_TILE_DIM + threadIdx.y;
 
-  for (int j = 0; j < TRANS_TILE_DIM; j += BLOCK_ROWS)
+  for (int j = 0; j < TRANS_TILE_DIM; j += BLOCK_ROWS) {
+    if (x >= m || (y + j) >= n) continue;
```

Comment:
x >= m, this condition never changes inside the loop. I wonder if j >= n - y is faster than y + j <= n as n - y is constant.

Reply:
will try, thx for the comments.

Reply:
In general, it is better to avoid for/if in GPU kernels. @ytaous as we talked offline, can you create an ADO to track "TODO" you described in PR to utilize Transpose3DKernel in onnxruntime/onnxruntime/core/providers/cuda/tensor/transpose_impl.cu?